elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryInterruptThread.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.UUID;
+import java.util.concurrent.Future;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.request.EntityProjection;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Runnable thread for terminating AsyncQueryThread executing
+ * beyond the max run time and update status.
+ */
+@Slf4j
+public class AsyncQueryInterruptThread implements Runnable {
+
+    private Elide elide;
+    private Future<?> task;
+    private UUID id;
+    private Date submittedOn;
+    private int interruptTime;
+
+    public AsyncQueryInterruptThread(Elide elide, Future<?> task, UUID id, Date submittedOn, int interruptTime){
+        log.debug("New Async Query Interrupt thread created");
+        this.elide = elide;
+        this.task = task;
+        this.id = id;
+        this.submittedOn = submittedOn;
+        this.interruptTime = interruptTime;
+    }
+
+    @Override
+    public void run() {
+        interruptQuery();
+    }
+
+    /**
+     * This is the main method which interrupts the Async Query request, if it has executed beyond
+     * the maximum run time.
+     */
+    protected void interruptQuery() {
+        try {
+            long interruptTimeInMillies = interruptTime * 60 * 1000;
```

Comment:
Let's make these configurable.

Reply:
interruptTime is set when the elide instance comes up based on the properties/settings set by the user in application.yaml/settings.java and it gets applied to all the queries. If the user does not set it, its defaulted to 60 minutes.  Do you mean it should be an attribute in AsyncQuery and every query should have its own timeout provided by the user? if not provided, we default it.